src/Search/Algolia/Index.php

```diff
@@ -63,7 +65,14 @@ public function update()
 
     public function getIndex()
     {
-        return $this->client->initIndex($this->name);
+        $index_created = ! $this->exists();
```

Comment:
I didn't know where all the getIndex might be called and I thought it might be an unintended side-effect if every time you call getIndex instead of just getting the index it also updated the settings?  That variable is there to ensure the setSettings is only called if the initIndex created the index.

Reply:
but why not `if (! $this->exists())`?

Reply:
Because if the index might get created on the next line (I think). He only wants to `setSettings` if it didn't already exist.

Reply:
Yes I can't call `setSettings` until the `initSettings` is called so it needs to know if `initSettings` created the index but only after it possibly created it.